conference/src/main/java/org/spine/samples/lobby/conference/ConferenceServiceImpl.java

```diff
@@ -0,0 +1,206 @@
+        responseObserver.onNext(response);
+        responseObserver.onCompleted();
+    }
+
+    private Conference updateFromConferenceInfo(Conference target, EditableConferenceInfo info) {
+        return target.toBuilder()
+                     .setName(info.getName())
+                     .build();
+    }
+
+    @Override
+    public void publish(PublishConferenceRequest request, StreamObserver<PublishConferenceResponse> responseObserver) {
+        final Conference persistedConference = conferenceRepository.load(request.getId());
+        //TODO:2016-04-19:andrii.loboda: handle if no conference to publish
+        if (!persistedConference.getIsPublished()) {
+            final Conference publishedConference = persistedConference.toBuilder()
+                                                                      .setIsPublished(true)
+                                                                      .build();
+
+            sendConferencePublishedEvent(publishedConference);
+            final PublishConferenceResponse response = PublishConferenceResponse.newBuilder()
+                                                                                .setId(persistedConference.getId())
+                                                                                .build();
+            responseObserver.onNext(response);
+            responseObserver.onCompleted();
+
+        } else {
+
+            //TODO:2016-04-19:andrii.loboda: handle if conference is already published
+        }
+
+    }
+
+    protected BoundedContext getBoundedContext() {
+        //TODO:2016-04-14:andrii.loboda:  move to singleton and provide event bus, command bus, etc.
+        return BoundedContext.newBuilder()
+                             .build();
+    }
+
+    protected ConferenceRepository getRepository() {
+        return new ConferenceRepository();
+    }
+
+
+    private void sendConferenceCreatedEvent(Conference createdConference) {
+        final ImmutableList.Builder<Event> result = ImmutableList.builder();
+        final ConferenceCreated conferenceCreatedEvent = EventFactory.conferenceCreated(createdConference);
+        final EventContext conferenceEventContext = EventUtils.createConferenceEventContext(createdConference.getId());
+        result.add(Events.createEvent(conferenceCreatedEvent, conferenceEventContext));
+
+        sendEvents(result.build());
+    }
+
+    private void sendConferenceUpdatedEvent(Conference updatedConference) {
+        final ImmutableList.Builder<Event> result = ImmutableList.builder();
+        final ConferenceUpdated conferenceUpdatedEvent = EventFactory.conferenceUpdated(updatedConference);
+        final EventContext conferenceEventContext = EventUtils.createConferenceEventContext(updatedConference.getId());
+        result.add(Events.createEvent(conferenceUpdatedEvent, conferenceEventContext));
+//TODO:2016-04-19:andrii.loboda: looks weird
```

Comment:
Yeah, this is weird indeed. `sendEvents` method is always used with the one-element-list. Please convert it to varargs method so that we don't have to create a list.